### PR TITLE
fix disc cache key increasing infinity

### PR DESCRIPTION
**Pull request by Yancey1989**

### CODE REVIEW

File: pytorch_blade/pytorch_blade/pybind.cpp [hunk 1]
```diff
@@ -70,6 +70,12 @@ void initModules<false>(py::module& m) {
           "jit_pass_propagate_input_shapes",
           &torch::blade::PropagateInputShapes)
       .def("jit_pass_constant_propagation", &torch::blade::ConstantPropagation);
+  m.def("hash_data", [&](std::string value) -> size_t {
```
Comment (qiuxiafei) [inline pytorch_blade/pytorch_blade/pybind.cpp:73]: pass `value` as reference?

File: pytorch_blade/torch_blade/mlir/hash.py [hunk 2]
```diff
@@ -13,14 +13,9 @@
 import numpy as np
 import hashlib
 from torch_blade._torch_blade import _backends
+from torch_blade import hash_data, hash_combine
 HASH_SEED = 1
 
-def data_hash(value: str) -> int:
```
Comment (qiuxiafei) [inline pytorch_blade/torch_blade/mlir/hash.py:18]: nit: removed for performance reason?

Comment (Yancey1989) [reply pytorch_blade/torch_blade/mlir/hash.py:18]: hashlib.sha256 returns a 64  character hash string that is too long and increased infinity after `hash_combine`.

File: pytorch_blade/torch_blade/mlir/hash.py [hunk 3]
```diff
@@ -31,10 +26,10 @@ def get_graph_hash(graph):
...
-        hash_value = hash_combine(hash_value, data_hash(val_info.dtype))
-        hash_value = hash_combine(hash_value, data_hash(val_info.device))
-        hash_value = hash_combine(hash_value, data_hash(str(rank)))
+        hash_value = hash_combine(hash_value, hash_data(val_info.dtype))
+        hash_value = hash_combine(hash_value, hash_data(val_info.device))
+        hash_value = hash_combine(hash_value, hash_data(str(rank)))
     # hash node info
     for n in nodes:
-        hash_value = hash_combine(hash_value, data_hash(n.kind()))
+        hash_value = hash_combine(hash_value, hash_data(n.kind()))
```
Comment (zzpmiracle) [inline pytorch_blade/torch_blade/mlir/hash.py:34]: Can we add input dtype of nodes? Some models in TorchBench meet this problem that graph input is int, and model are half or float
```
        for input in n.inputs():
                hash_value = hash_combine(hash_value, hash_data(input.type().str()))
```

Comment (Yancey1989) [reply pytorch_blade/torch_blade/mlir/hash.py:34]: That's a good point! Thanks.

**Review by zzpmiracle — APPROVED** [review]
LGTM.

**Pull request merged**